server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -294,12 +294,18 @@ Metadata upgradeMetadataForNode(Metadata metadata, IndexMetadataVerifier indexMe
      */
     static Metadata upgradeMetadata(Metadata metadata, IndexMetadataVerifier indexMetadataVerifier, MetadataUpgrader metadataUpgrader) {
         boolean changed = false;
+        IndexMetadataVerifier.MetadataVerificationMode metadataVerificationMode = metadata.clusterUUIDCommitted() // If true, a cluster
+                                                                                                                  // already exists
+            ? IndexMetadataVerifier.MetadataVerificationMode.NO_UPGRADE
+            : IndexMetadataVerifier.MetadataVerificationMode.WITH_UPGRADE;
```

Comment:
If a cluster exists or not in the metadata read from disk is not relevant. 
As discussed, in the general case we'll be doing a rolling upgrade to (re-)join an existing cluster. Whatever cluster state exists locally, if will be discarded as we're not master when joining that cluster. When starting a node, we always want to skip archiveOrDeleteBrokenIndexSettings in IndexMetadataVerifier.